interview-questions.md

```diff
@@ -7,65 +7,113 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 4. What are the 5 HTTP verbs? What are they important?
 
-  Your answer:
+  Your answer: GET POST PUT PATCH DELETE. They relate to the CRUD operations you can do in a database. The HTTP verbs are ways to retrieve (GET) data, (POST) update or upload data, (PUT) is done to replace or change data, (PATCH) is another way of changing or replacing data although it is frowned upon it allows you to change a singular piece of data rather than a whole set of data which is normally what happens when you use (PUT). And finally (DELETE) is done to delete data
 
-  Researched answer:
+  Researched answer: The HTTP verbs comprise a major portion of our “uniform interface” constraint and provide us the action counterpart to the noun-based resource. The primary or most-commonly-used HTTP verbs (or methods, as they are properly called) are POST, GET, PUT, PATCH, and DELETE. These correspond to create, read, update, and delete (or CRUD) operations, respectively. There are a number of other verbs, too, but are utilized less frequently. Of those less-frequent methods, OPTIONS and HEAD are used more often than others.
 
 
 
 5. What is object-relational mapping?
 
-  Your answer:
+  Your answer: Object-relational mapping also known as ORM is used to convert database information into readable and understanable data into our code editor. It makes it much easier to retrieve data from a database.
 
   Researched answer:
+  ORMs generate objects which map to tables in the database virtually. Once these objects are up then coders can easily work to retrieve, manipulate or delete any field from the table without paying much attention to language specifically. It supports writing the complex long SQL queries in a simpler way.
 
+  ORMs will shield your application from SQL injection attacks since the framework will filter the data for you! ORMs provide the concept of Database Abstraction which makes switching databases easier and creates a consistent code base for your application.
 
 
 6. What is a gem?
 
-  Your answer:
+  Your answer: Installing gem lets you install rspec
 
-  Researched answer:
+  Researched answer: 
+  RubyGems is a package manager for the Ruby programming language that provides a standard format for distributing Ruby programs and libraries (in a self-contained format called a "gem"), a tool designed to easily manage the installation of gems, and a server for distributing them.
 
 
 
 7. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are assigned to each value inside of an object?
```

Comment:
Primary keys are generated by the database to identify each record/instance of a model/table